src/commands/workflow/compile.ts

```diff
@@ -0,0 +1,65 @@
+import {existsSync, readFileSync} from 'fs'
+import {dirname, join} from 'path'
+
+import Command from '../../root-command'
+import * as compile from '../../utils/compiler'
+import ServiceCompile from '../service/compile'
+import ServiceCreate from '../service/create'
+import ServiceStart from '../service/start'
+
+export default class WorkflowCompile extends Command {
+  static description = 'Compile a workflow'
+
+  static flags = {
+    ...Command.flags
+  }
+
+  static args = [{
+    name: 'WORKFLOW_FILE',
+    description: 'Path of a workflow file'
+  }]
+
+  async run(): Promise<any> {
+    const {args} = this.parse(WorkflowCompile)
+    const definition = await compile.workflow(readFileSync(args.WORKFLOW_FILE), async (instanceObject: any) => {
+      if (instanceObject.instanceHash) {
+        return instanceObject.instanceHash
+      }
+      if (instanceObject.service) {
+        return this.serviceToInstance(instanceObject.service)
+      }
+      if (instanceObject.src) {
+        return this.sourceToInstance(args.WORKFLOW_FILE, instanceObject.src)
+      }
+      throw new Error('instance resolution error')
+    })
+    this.styledJSON(definition)
+    this.spinner.stop()
+    return definition
+  }
+
+  async sourceToInstance(dir: string, src: string): Promise<string> {
+    const directory = join(dirname(dir), src)
+    const definition = await ServiceCompile.run([existsSync(directory) ? directory : src, '--silent'])
```

Comment:
Could we plug the `--silent` flag of the command `workflow:compile`?
So if not `--silent` is passed to `workflow:compile`, we can follow what the CLI is doing (eg: "deploying service XXX", etc..). Same for starting the instance..

Reply:
the problem is, for now, the compile needs to print the result to be readable for the command create so having user interaction will "pollute" the output.
This can be changed later and introduce maybe a flag `[no-]interactive` or something like that